modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItBuildIndexTest.java

```diff
@@ -275,6 +284,10 @@ private static void checkIndexBuild(int partitions, int replicas, String indexNa
                     IgniteStringFormatter.format("p={}, nodes={}", entry.getKey(), entry.getValue())
             );
         }
+
+        waitForCondition(() -> isIndexAvailable(INDEX_NAME), 10_000);
+
+        waitForReadTimestampThatObservesMostRecentCatalog();
```

Comment:
there is a javadoc that explains this.

> Waits for some amount of time so that read-only transactions can observe the most recent version of the catalog.

Reply:
Added an additional comment.